app/Repositories/Doctrine/AccountRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repositories\Doctrine;
+
+use App\Database\Entities\Account;
+use App\Database\Exceptions\InvalidTypeSubscriptionException;
+use App\Repositories\Doctrine\ORM\AbstractRepository;
+use App\Repositories\Interfaces\AccountRepositoryInterface;
+
+final class AccountRepository extends AbstractRepository implements AccountRepositoryInterface
+{
+    /**
+     * Find by subscription type.
+     *
+     * @param string $subscriptionType
+     *
+     * @return mixed[]
+     */
+    public function findBySubscriptionType(string $subscriptionType): array
+    {
+        try {
+            if (!\in_array($subscriptionType, Account::SUBSCRIPTION_TYPES, true)) {
+                throw new InvalidTypeSubscriptionException();
+            }
+
+            return $this->repository->findBy(['subscriptionType' => $subscriptionType]);
+        } catch (InvalidTypeSubscriptionException $ex) {
+            return ['error' => $ex->getErrorCode()];
```

Comment:
Here, since this is `Account` repository find by, the expected return should be an array of Accounts or empty array only. If there is an error, we just let the application handle it in the Exceptions handler class.

So here, `return $this->repository->findBy(['subscriptionType' => $subscriptionType])` is enough ;)